### DISCUSSION TITLE
Whether downloading and uploading images through API interface is supported？

### DISCUSSION THREAD

**Discussion by fzwu** [goharbor/harbor#20300] [Ideas]
I use. NET 6 to develop applications and deploy them to the K8S cluster. 
Because the operation and maintenance staff want to upload and download the update through the UI interface, they need to upload and download the image through the API interface, but the program runs in the container and cannot execute the docker command. 
Is it possible to open relevant APIs for image upload? 
For example: 
        download tar package of image and upload tar package？

**Comment by microyahoo** [comment]
harbor provide [go-client](https://github.com/goharbor/go-client), but you can generate corresponding language through [swagger](https://github.com/goharbor/go-client/blob/main/api/v2.0/swagger.yaml).

**Comment by chlins** [comment]
Currently harbor does not have the direct API to upload or download the image because the behavior is not defined in the OCI spec, but per your requirement you can consider to use the 3rd party OCI client package such as oras(golang ported) to help you implement the scenario.

**Comment by fzwu** [comment]
Okay, I'll try another way.
